### PR TITLE
Fix IDE0008 (use explicit type instead of var) to modernize the codebase

### PR DESCRIPTION
**Pull request by ghost**
Every change was done automatically by the IDE. This is being done to ensure the codebase meets modern C# standards. No behavioral changes.

### CODE REVIEW

File: osu.Game/Graphics/UserInterface/OsuSliderBar.cs [hunk 1]
```diff
@@ -129,10 +129,10 @@ protected override void OnUserChange(T value)
 
         private void playSample(T value)
         {
-            if (Clock == null || Clock.CurrentTime - lastSampleTime <= 50)
```
Comment (peppy) [inline osu.Game/Graphics/UserInterface/OsuSliderBar.cs]: were these changes also automated? could you tell us which plugin / refactoring tool you used to perform these?

**Comment by peppy** [comment]
Can you explain where these var changes were made? There are still many existing (potentially ambiguous) usages like:

image

where one may argue that not using `var` is more readable.

File: osu.Game.Rulesets.Osu/UI/OsuPlayfield.cs [hunk 2]
```diff
@@ -86,7 +86,7 @@ public override void PostProcess()
 
         private void onNewResult(DrawableHitObject judgedObject, JudgementResult result)
         {
-            if (!judgedObject.DisplayResult || !DisplayJudgements.Value)
+            if (!(judgedObject.DisplayResult && DisplayJudgements.Value))
```
Comment (iiSaLMaN) [inline osu.Game.Rulesets.Osu/UI/OsuPlayfield.cs]: how is this related?

File: osu.Game.Rulesets.Osu/Edit/OsuSelectionHandler.cs [hunk 3]
```diff
@@ -13,7 +13,7 @@ public class OsuSelectionHandler : SelectionHandler
     {
         public override void HandleDrag(SelectionBlueprint blueprint, DragEvent dragEvent)
         {
-            foreach (var h in SelectedHitObjects.OfType<OsuHitObject>())
+            foreach (OsuHitObject h in SelectedHitObjects.OfType<OsuHitObject>())
```
Comment (iiSaLMaN) [inline osu.Game.Rulesets.Osu/Edit/OsuSelectionHandler.cs]: is there a problem with using `var` here? it can be clearly seen that the type is `OsuHitObject` because of the filtering `Enumerable.OfType<FooType>()`

File: osu.Game/Database/MutableDatabaseBackedStore.cs [hunk 4]
```diff
@@ -131,15 +131,15 @@ public void PurgeDeletable(Expression<Func<T, bool>> query = null)
...
 
-                var lookup = context.Set<T>().Where(s => s.DeletePending);
+                IQueryable<T> lookup = context.Set<T>().Where(s => s.DeletePending);
 
                 if (query != null) lookup = lookup.Where(query);
 
                 lookup = AddIncludesForDeletion(lookup);
 
-                var purgeable = lookup.ToList();
+                List<T> purgeable = lookup.ToList();
```
Comment (iiSaLMaN) [inline osu.Game/Database/MutableDatabaseBackedStore.cs]: also here, we're converting the `IQueryable<T>` to list, isn't it obvious that the type would be `List<T>`

File: osu.Game/Graphics/Backgrounds/Triangles.cs [hunk 5]
```diff
@@ -228,7 +228,7 @@ public override void Draw(Action<TexturedVertex2D> vertexAction)
                 {
                     var offset = triangle_size * new Vector2(particle.Scale * 0.5f, particle.Scale * 0.866f);
 
-                    var triangle = new Triangle(
+                    Triangle triangle = new Triangle(
```
Comment (iiSaLMaN) [inline osu.Game/Graphics/Backgrounds/Triangles.cs]: we're creating an instance of `Triangle`, is there a reason not to use `var` here as well?

**Review by frenzibyte — CHANGES_REQUESTED** [review]

File: osu.Game.Rulesets.Osu/UI/OsuPlayfield.cs (hunk 2 above)
Comment (pi1024e) [reply osu.Game.Rulesets.Osu/UI/OsuPlayfield.cs]: Undid that.

File: osu.Game.Rulesets.Osu/Edit/OsuSelectionHandler.cs (hunk 3 above)
Comment (pi1024e) [reply osu.Game.Rulesets.Osu/Edit/OsuSelectionHandler.cs]: To make it clear it is a hit object, because "var h" is not a clear enough

File: osu.Game/Database/MutableDatabaseBackedStore.cs (hunk 4 above)
Comment (pi1024e) [reply osu.Game/Database/MutableDatabaseBackedStore.cs]: There are many subclasses of List<T>, so yes, it would not be as obvious as you think.

File: osu.Game.Rulesets.Osu/Edit/OsuSelectionHandler.cs (hunk 3 above)
Comment (peppy) [reply osu.Game.Rulesets.Osu/Edit/OsuSelectionHandler.cs]: it's made clear by the source in this case.

File: osu.Game/Graphics/Backgrounds/Triangles.cs (hunk 5 above)
Comment (pi1024e) [reply osu.Game/Graphics/Backgrounds/Triangles.cs]: The object being returned is a Triangle, so making it Triangle instead of var makes it stand out.

Comment (iiSaLMaN) [reply osu.Game/Graphics/Backgrounds/Triangles.cs]: this makes no sense, you're returning a `Triangle`, we would expect `var` to be `Triangle`?

File: osu.Game.Rulesets.Catch/Beatmaps/CatchBeatmapConverter.cs [hunk 6]
```diff
@@ -22,11 +22,11 @@ public CatchBeatmapConverter(IBeatmap beatmap)
 
         protected override IEnumerable<CatchHitObject> ConvertHitObject(HitObject obj, IBeatmap beatmap)
         {
-            var curveData = obj as IHasCurve;
-            var positionData = obj as IHasXPosition;
-            var comboData = obj as IHasCombo;
-            var endTime = obj as IHasEndTime;
-            var legacyOffset = obj as IHasLegacyLastTickOffset;
+            IHasCurve curveData = obj as IHasCurve;
```
Comment (iiSaLMaN) [inline osu.Game.Rulesets.Catch/Beatmaps/CatchBeatmapConverter.cs]: it's casted to `IHasCurve`, why is it not var...?

**Comment by iiSaLMaN** [comment]
Please test your changes locally before pushing to reduce failed builds, you can do that by running `build.ps1` for windows or `build.sh` for mac/linux

**Comment by peppy** [comment]
Please do not force push after making a PR.

File: osu.Game.Rulesets.Osu/Edit/OsuSelectionHandler.cs (hunk 3 above)
Comment (peppy) [reply osu.Game.Rulesets.Osu/Edit/OsuSelectionHandler.cs]: please don't resolve conversations

**Comment by peppy** [comment]
This is mostly non-required changes. Our current policy on `var` usage is as follows:

- If the type can be discerned from the line (ie. there is `as Type` or `new Type`) we use `var`
- If the type is a primitive type, we don't use var.

Anywhere else we use var or not var (it's personal preference). Generally we use `var` for longer type names or to avoid type repetition.

I'm going to close this PR. Feel free to open another one following the above guidelines if you find any that you feel are out of line.
